src/main/java/com/cloudbees/jenkins/support/impl/RunDirectoryComponent.java

```diff
@@ -97,6 +91,26 @@ public boolean understandsSymlink() {
         }
     }
 
+    private static String getBuildRelativePath(@NonNull Run item, @NonNull File itemRootDir) {
+
+        // See Jenkins#isDefaultBuildDir that has restriction and cannot be used outside Jenkins core
```

Comment:
I think it is `Jenkins.getBuildDirFor` you want to look at. What if `Jenkins.buildsDir` is set to, e.g., `${ITEM_ROOTDIR}/all-the-builds`?